src/torrent.cpp

```diff
@@ -10554,22 +10550,22 @@ namespace libtorrent
 
 		// activity time
 #ifndef TORRENT_NO_DEPRECATE
-		st->finished_time = finished_time();
+		st->finished_time = int(duration_cast<seconds>(finished_time()).count());
 		st->active_time = active_time();
 		st->seeding_time = seeding_time();
 
-		st->time_since_upload = m_last_upload == (std::numeric_limits<std::int16_t>::min)() ? -1
-			: clamped_subtract_u16(m_ses.session_time(), m_last_upload);
-		st->time_since_download = m_last_download == (std::numeric_limits<std::int16_t>::min)() ? -1
-			: clamped_subtract_u16(m_ses.session_time(), m_last_download);
+		st->time_since_upload = int(total_seconds(m_last_upload
+			- m_ses.session_start_time()));
```

Comment:
I think you meant to say ``aux::time_now() - m_last_upload``, no? (and same for ``m_last_download``)